<?xml version="1.0"?>
<launch>
    <!-- Gazebo options -->
    <arg name="gazebo_gui" default="true" doc="Start Gazebo GUI"/>
    <arg name="paused" default="false" doc="Start Gazebo paused"/>
    <arg name="world_name" default="worlds/empty.world" doc="Gazebo world file"/>
    <arg name="world_pose" default="-x 0 -y 0 -z 0 -R 0 -P 0 -Y 0" doc="Pose to spawn the robot at"/>
    <arg name="initial_joint_positions" default=" -J left_arm_joint_1 -0.0176 -J left_arm_joint_2 0.653 -J left_arm_joint_3 -0.0176 -J left_arm_joint_4 0.0706 -J left_arm_joint_5 -0.7589 -J left_arm_joint_6 0 -J right_arm_joint_1 -0.1412 -J right_arm_joint_2 1.0589 -J right_arm_joint_3 -0.7413 -J right_arm_joint_4 0.0882 -J right_arm_joint_5 1.6413 -J right_arm_joint_6 0.0353" doc="Initial joint configuration of the robot"/>
    <arg name="mfi" default="false" />
    <arg name="async" default="false" />
    <arg name="load_tpg" default="false" />
    <arg name="load_adg" default="false" />

    <!-- Start the robot state publisher, simulation, and controler -->
    <!-- <include file="$(find dual_gp4_moveit_config)/launch/demo_gazebo.launch">
    </include> -->

    <include file="$(find dual_gp4_moveit_config)/launch/demo.launch"/>

    <!-- Run the main MoveIt executable with trajectory execution -->
    <!-- <include file="$(find dual_gp4_moveit_config)/launch/move_group.launch">
        <arg name="allow_trajectory_execution" value="false" />
        <arg name="moveit_controller_manager" value="fake" />
        <arg name="info" value="true" />
        <arg name="pipeline" value="ompl" />
        <arg name="load_robot_description" value="true" />
    </include> -->

    <!-- Start moveit_rviz with the motion planning plugin -->
    <!-- <include file="$(find dual_gp4_moveit_config)/launch/moveit_rviz.launch">
        <arg name="rviz_config" value="$(find dual_gp4_moveit_config)/launch/moveit.rviz" />
    </include> -->

    <!-- Start the planning node -->
    
    <node launch-prefix="xterm -e gdb --args" name="lego_node" pkg="mr_planner" type="lego_node" output="screen">
            <param name="log" value="debug"/>
        <param name="fullorder_targets_filename" value="$(find mr_planner)/assembly_steps/stairs_top.csv" />
        <param name="planner_type" value="PrioritizedPlanning"/>
        <param name="config_fname" value="$(find lego_manipulation)/config/user_config.json"/>
        <param name="root_pwd" value="$(find lego_manipulation)/"/>
        <param name="output_dir" value="$(find mr_planner)/outputs/lego"/>
        <param name="group_name_0" value="/left_arm" />
        <param name="group_name_1" value="/right_arm" />
        <param name="mfi" value="$(arg mfi)" />
        <param name="async" value="$(arg async)" />
        <param name="load_tpg" value="$(arg load_tpg)" />
        <param name="load_adg" value="$(arg load_adg)" />
    </node>
    
    <node name="left_arm" pkg="mr_planner" type="yk_robot" output="screen">
        <param name="group_name" value="left_arm" />
    </node>

    <node name="right_arm" pkg="mr_planner" type="yk_robot" output="screen">
        <param name="group_name" value="right_arm" />
    </node>


</launch>